vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,403 @@
+
+
+## Exploring your data {.modelData}
+
+<!--
+#TODO Non-stationarity, longer time scales, multi-modality from known etiologies
+-->
+
+It is important to understand the data being passed to `epinowcast` and have some ideas regarding the data generating process for your data.
+For instance, if there are significant changes in your reporting triangle, this could manifest as multi-modality in the posterior distribution (i.e., Stan is trying to fit two different reporting delays with the same parameters).
+These different reporting regimes could result in long model run times, very wide posteriors, and ultimately nonsensical inferences.
+Given the nuances of your data, you may need to identify and fit different regimes of your data.
+These problems can also manifest over longer timescales, so while more data are generally better, too much data can result in ill-fitting models.
+
+
+## Common failure modes
+
+### My model takes too long to run
+
+One of the more common issues with Bayesian inference is that the sampler may take a long time to run.
+This can occur because the sampler is having difficulty exploring the posterior parameter space given the model and your data.
+You can a few simple steps to assist with this:
+
+
+1. Increase the [`adapt_delta`](.adaptDelta) argument to 15 which will allow longer trajectories to used for steps when sampling. This is especially important if the `per_at_max_treedepth` value is high in the returned CmdStan diagnostics.
+2. Use more [informative priors](.settingPriors). 
+3. Try to simplify the model first to understand potential degeneracies.
+4. Reduce the amount of data you are trying to fit. This might include reducing the duration of time overwhich you are trying to fit. Once you can confirm that the model will fit your data, it may just be a question of computational power when fitting the larger data. 
+
+
+### Divergent transitions 
+
+CmdStan may alert you that you have ["divergent transitions"](https://mc-stan.org/docs/reference-manual/divergent-transitions.html) detected during your model fit.
+
+```r
+#>1: There were 32 divergent transitions after warmup.
+```
+
+A high number of divergent transitions indicates that the model outputs are not to be trusted.
+The first step in addressing this is to increase the [`adapt_delta`](#adaptDelta) argument to a numeric near one (e.g., 0.99 or 0.999).
+This will allow for a smaller step size and may result in more realiable inferences.
+Importantly, if the number of divergent transitions remains high, your model may be misspecified given the data that are available.
+At this point it may be worthwhile to investigate if the model paramterisation is correct for the data you have available.
+
+
+### My `r_hat`s are high and by `ess`s are low
+
+High values of `r_hat`, the Gelman-Rubin statistic used to measure converge, indicates a lack of convergence.
+Values greater than 1.05 are considered unreliable.
```

Comment:
This advice conflicts with the standard advice in the Stan help pages of $\hat R \ge 1.01$ indicating non-convergence. I think you're probably right here, but it would be good to address this discrepancy explicitly.

Reply:
This does match the `rstan` guidance but maybe its out of date: https://mc-stan.org/rstan/reference/Rhat.html#:~:text=Both%20bulk%2DESS%20and%20tail,respective%20posterior%20quantiles%20are%20reliable.

Reply:
If you're comfortable citing the Stan forums in here, [this thread between two Stan dev](https://discourse.mc-stan.org/t/summarising-rhat-values-over-multiple-variables-fits/23957/5)s is a really nice discussion. I'd also cite to [the $\hat R$ paper](https://projecteuclid.org/journals/bayesian-analysis/volume-16/issue-2/Rank-Normalization-Folding-and-Localization--An-Improved-R%cb%86-for/10.1214/20-BA1221.full) which has the original guidance for the 1.01 threshold.

Reply:
Thanks for hunting these down @zsusswein @seabbs . My Rhat < 1.05 dates my training 👴 . I'll also add this [case study](https://avehtari.github.io/rhat_ess/ess_comparison.html) which relates to the $\hat R$ paper and integrate these in the next round (hopefully tomorrow afternoon eastern time)